### DISCUSSION TITLE
Define a component outside of an OAM application

### DISCUSSION THREAD

**Discussion by nitagarwal** [kubevela/kubevela#5877] [Q&A]
I am looking for a way to define an application component in it's own yaml file and then reference it in OAM application spec. Is it possible with OAM?

**Comment by Somefive** [comment]
It is not possible for the current KubeVela. Before we have tried that way but later we drop that usage. A separate component could be a bit confusing, especially when sharing across multiple applications.

I'm curious in what cases do you need to reference a component outside?
